ProcessOptimizer/model_systems/color_pH.py

```diff
@@ -0,0 +1,339 @@
+        for row in reader:
+            data.append(row)
+
+    # Convert data to numpy array for easier manipulation
+    data = np.array(data)
+
+    # Get the indices of the target columns
+    target_indices = [headers.index(col) for col in ['Well','L','A','B']]
+
+    # Extract the data for the target columns
+    target_data = data[:, target_indices]
+    
+    # Take the row in which the 'Well' column matches the target well
+    target_row = target_data[target_data[:,0] == target_well]
+    
+    #Transform the values in 'L','A','B' to floats
+    target_row = target_row[0][1:].astype(float)
+    
+    #Return the values in 'L','A','B' as a dictionary
+    return {'L': target_row[0], 'a': target_row[1], 'b': target_row[2]}
+
+
+def color_difference(file_name, well1, well2):
+    '''
+    file_name: name of the csv file
+    well1: well number of the first well
+    well2: well number of the second well
+    
+    returns: delta-e value between the two wells
+    '''
+    #Find the lab color using color_finder and calculate the difference between the two colors. use the method delta_E from the colour package.
+    color1 = color_finder_dict(file_name, well1)
+    color2 = color_finder_dict(file_name, well2)
+    #return colour.delta_E(color1, color2, method='CIE 2000')
+    return delta_e_2000(color1, color2, verbose=False, test=False, formula='Bruce')
+
+
+def score(coordinates: List[int], evaluation_target='F8'):
+    """
+    coordinates: list of coordinates for a single experiment
+    evaluation_target: well number of the target well. Default is F8 (because green is good for the eyes)
+    
+    returns: delta-e value between the two wells
+    
+    Takes a list of coordinates for a single experiment and compare to the color in target cell. It then returns the delta-e value between the two wells.
+    With the default value of F8 as target, the right recipy will be 50% acid and 30uL Indicator [50,30].
+    """
+    # make a relative path to the data file called color_pH_data.csv. the file should be in a subfolder called data. the path should be relative to the file that is calling this function.
+    package_directory = os.path.dirname(os.path.abspath(__file__))
+    file_name = os.path.join(package_directory, 'data', 'color_pH_data.csv')
+    data_lookup_position = find_closest_match(file_name, coordinates, ['percent_acid', 'Indicator'])[0]
+    evaluation = color_difference(file_name, data_lookup_position, evaluation_target)
+    return evaluation
+    
+#Create model system
+color_pH_no_noise = ModelSystem(
+    score,
+    space = [Integer(30, 85, name='percent_acid'),
+             Integer(5, 40, name='Indicator'),
+            ],
+    noise_model = None,
+    true_min = 0,
+)
+
+color_pH = ModelSystem(
+    score,
+    space = [Integer(30, 85, name='percent_acid'),
+             Integer(5, 40, name='Indicator'),
+            ],
+    noise_model = {"model_type": "proportional", "noise_size": 0.1},
```

Comment:
Let's discuss whether the noise should be proportional.

My naïve argument would be that the measured data contains all of the noise we want, so let's not add more, but I could se arguments against this.

Reply:
I'm up for any type of noise. The reason for its additio is just to "create awareness" about the ability to work with noise in the package